### PR TITLE
capture map

### PR DESCRIPTION
**Pull request by EllAchE**
- fixes to move logic
- add ep test
- improve ep test
- add streaks and revenge kills

### CODE REVIEW

File: cjsmin/src/chess.ts [hunk 1]
```diff
@@ -133,6 +133,26 @@ const SQUARE_TO_STARTING_POSITION_MAP = {
...
+  "a3", "b3", "c3", "d3", "e3", "f3", "g3", "h3",
+  "a4", "b4", "c4", "d4", "e4", "f4", "g4", "h4",
+  "a5", "b5", "c5", "d5", "e5", "f5", "g5", "h5",
+  "a6", "b6", "c6", "d6", "e6", "f6", "g6", "h6",
+  "a7", "b7", "c7", "d7", "e7", "f7", "g7", "h7",
+  "a8", "b8", "c8", "d8", "e8", "f8", "g8", "h8"
+]
+
+// prettier-ignore
+export const ALL_UNAMBIGUOUS_PIECE_SYMBOLS: UnambiguousPieceSymbol[] = [
```
Comment (bennyrubanov) [inline cjsmin/src/chess.ts:149]: why were these added?

File: cjsmin/src/chess.ts [hunk 2]
```diff
@@ -1272,11 +1292,18 @@ export class Chess {
...
+                unambiguousSymbol: this._board[to].unambiguousSymbol,
               },
               BITS.CAPTURE
             );
           } else if (to === this._epSquare) {
+            // for the unambiguous symbol we can get the file and casing
+            const file = algebraic(to).charAt(0);
+            let uas = 'p' + file;
+            if (us === BLACK) {
+              uas = uas.toUpperCase();
```
Comment (bennyrubanov) [inline cjsmin/src/chess.ts:1304]: black should be lowercase?

**Review by bennyrubanov — APPROVED** [review]

File: cjsmin/src/chess.ts (hunk 2 above)
Comment (EllAchE) [reply cjsmin/src/chess.ts:1304]: think you saw but this is an undoMove, so us = BLACK means them white

File: cjsmin/src/chess.ts (hunk 1 above)
Comment (EllAchE) [reply cjsmin/src/chess.ts:149]: usages in a couple spots

File: cjsmin/src/chess.ts (hunk 2 above)
Comment (bennyrubanov) [reply cjsmin/src/chess.ts:1304]: yep understand now thx
